crates/core/src/config.rs

```diff
@@ -35,6 +31,24 @@ pub struct MetadataFile {
 }
 
 impl MetadataFile {
+    pub fn new(edition: &str) -> Self {
+        let mut current_version: semver::Version = env!("CARGO_PKG_VERSION").parse().unwrap();
+        // set the patch version of newly-created metadata files to 0 -- v1.0.0
+        // set `cmp.patch = Some(file_version.patch)` when checking version
+        // compatibility, meaning it won't be forwards-compatible with a
+        // database claiming to be created on v1.0.1, even though that should
+        // work. This can be changed once we release v1.1.0, since we don't
+        // care about its DBs being backwards-compatible with v1.0.0 anyway.
+        if let semver::Version { major: 1, minor: 0, .. } = current_version {
+            current_version.patch = 0;
+        }
```

Comment:
what do you think about just unconditionally setting `patch` to 0? That way, we don't need a fancier comparator.

Reply:
The comparator isn't really fancier than it was before, it's just swapping `Some(patch)` for `None` -- and I think we may want to know if the database has been used on a specific patch version in the future, or at least there's no reason not to store that metadata in case it's ever needed.